### PR TITLE
allow skipping precompilation during development via ENV var

### PR DESCRIPTION
**Pull request by jkrumbiegel**

### CODE REVIEW

File: CairoMakie/src/CairoMakie.jl [hunk 1]
```diff
@@ -1,3 +1,7 @@
+if get(ENV, "MAKIE_PRECOMPILE", "true") == "false"
+    __precompile__(false)
+end
```
Comment (t-bltg) [inline CairoMakie/src/CairoMakie.jl:3]: Doesn't have this to be set inside the module definition ?

Xref: https://github.com/timholy/SnoopCompile.jl/issues/324.

Comment (jkrumbiegel) [reply CairoMakie/src/CairoMakie.jl:3]: It's documented that it disables precompilation for the whole file. I think I was told on Slack to do it like this but not completely sure

Comment (t-bltg) [reply CairoMakie/src/CairoMakie.jl:3]: I see.
Nitpicking, but these can be condensed to `__precompile__(get(ENV, "MAKIE_PRECOMPILE", "true") == "true")`.

Comment (jkrumbiegel) [reply CairoMakie/src/CairoMakie.jl:3]: Thought about that but found it clearer to show that `__precompile__` is not called at all if the variable isn't set.

Comment (t-bltg) [reply CairoMakie/src/CairoMakie.jl:3]: Negating the logic would also allow to use any string (e.g. an Int which is shorter to type) to disable precompilation.

Comment (SimonDanisch) [reply CairoMakie/src/CairoMakie.jl:3]: maybe use `&&` then?

**Comment by timholy** [comment]
Note there is also now https://github.com/timholy/SnoopCompile.jl/pull/334 (not yet released though, want to try a PkgEval run first).

**Comment by SimonDanisch** [comment]
Decided here https://github.com/MakieOrg/Makie.jl/issues/2585#issuecomment-1381585230 to not continue this PR.

**Pull request closed without merging**